pkg/client/kubernetes/clientmap/internal/garden_clientmap.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```

Comment:
Isn't this more or less the same like the shoot clientmap? Can we reuse the code instead of duplicating it?

Reply:
`garden` and `shoot` clientmap are both using the `generic-clientmap`, which contains the logic of how to get, validate and invalidate clientsets.
Only the factory types are different and I don't see how this could be reused. The shoot clientmap uses the shoot object, it uses two different clients (garden client and seed client).